moveos/moveos-types/src/event_filter.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+// use move_bytecode_utils::module_cache::GetModule;
+use move_core_types::account_address::AccountAddress;
+// use move_core_types::identifier::Identifier;
+// use move_core_types::language_storage::StructTag;
+use crate::event::Event;
+use crate::h256::H256;
+use crate::move_types::type_tag_match;
+use move_core_types::language_storage::TypeTag;
+// use schemars::JsonSchema;
+use crate::object::ObjectID;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use serde_with::serde_as;
+
+#[serde_as]
+// #[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+#[derive(Eq, PartialEq, Clone, Debug)]
+// #[serde(rename = "Event", rename_all = "camelCase")]
+pub struct MoveOSEvent {
+    pub key: ObjectID,
+    /// Sender's address.
+    pub sender: AccountAddress,
+    /// Transaction hash
+    pub tx_hash: H256,
+    /// The number of messages that have been emitted to the path previously
+    pub sequence_number: u64,
+    // #[schemars(with = "String")]
+    // #[serde_as(as = "TypeTag")]
+    /// Move event type.
+    pub type_tag: TypeTag,
+    /// bcs bytes of the move event
+    pub event_data: Vec<u8>,
+    /// Parsed json value of the event data
+    pub parsed_json: Value,
+    /// UTC timestamp in milliseconds since epoch (1/1/1970)
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    // #[schemars(with = "Option<u64>")]
+    // #[serde_as(as = "Option<u64>")]
+    pub timestamp_ms: Option<u64>,
+    /// block height
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    // #[schemars(with = "Option<u64>")]
+    // #[serde_as(as = "Option<u64>")]
+    pub block_height: Option<u64>,
+    /// event index in the transaction events.
+    pub event_index: u32,
+}
+
+impl MoveOSEvent {
+    pub fn try_from(
+        event: Event,
+        tx_hash: H256,
+        timestamp_ms: Option<u64>,
+        block_height: Option<u64>,
+    ) -> Result<Self> {
+        let Event {
+            key,
+            sequence_number,
+            type_tag,
+            event_data,
+            event_index,
+        } = event;
+
+        //TODO how to store and derive sender address ?
+        let sender = AccountAddress::ZERO;
```

Comment:
The current implemention does not store the sender in the GUID, so we can not know the sender. If we want to know the sender, we need to store then sender to the Event.